get_all_manifests.sh

```
@@ -61,22 +67,69 @@ fi
-    local git_fetch="git fetch -q --depth 1 $repo"
 
     mkdir -p $dir
     pushd $dir &>/dev/null
     git init -q
-    # try tag first, avoid printing fatal: couldn't find remote ref
-    if ! $git_fetch refs/tags/$ref 2>/dev/null ; then
-        $git_fetch refs/heads/$ref
+
+    # Check if ref is in tracking format: branch@sha
+    if [[ $ref =~ ^([a-zA-Z0-9_./-]+)@([a-f0-9]{7,40})$ ]]; then
+        local branch="${BASH_REMATCH[1]}"
```

Shellcheck (0.11.0)

[warning] branch appears unused. Verify use (or export if used externally).

(SC2034)